### PR TITLE
SFINT-2878 Added Cypress functional tests

### PR DESCRIPTION
**Pull request by jeremierobert-coveo**
I added Cypress to run functional test in the repo.

Here some screenshot.

image

And here too

image

### CODE REVIEW

**Comment by louis-bompart** [comment]
Can you split the PR in two?
- The setup
- The tests

So we can distinguish the setup and the tests in our review (and upcoming /blame :P )

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "baseUrl": "http://localhost:8080",
```
Comment (louis-bompart) [inline cypress.json]: https?

File: cypress.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "reporter": "spec",
+    "viewportHeight": 600,
```
Comment (louis-bompart) [inline cypress.json]: Maybe FHD?

File: cypress.json (hunk 1 above)
Comment (jeremierobert-coveo) [reply cypress.json]: nope, c'est discourage by cypress

File: cypress.json (hunk 2 above)
Comment (jeremierobert-coveo) [reply cypress.json]: What would be the gain ?

**Comment by jeremierobert-coveo** [comment]
imo, the setup is pointless with at least a single test to test the test setup.
And those User actions tests where quick and simple to write.

File: cypress.json (hunk 2 above)
Comment (louis-bompart) [reply cypress.json]: I see magic numbers I propose new ones :P 
1920*1080 seems closer to real life application to me.

**Comment by louis-bompart** [comment]
> imo, the setup is pointless with at least a single test to test the test setup.
> And those User actions tests where quick and simple to write.

The thing is if your tests become an issue, we will have a single commit to revert without having to untangle what to keep, what to ditch.

It keeps things contained. Like, your setup should not be hard-linked with a feature

**Comment by jeremierobert-coveo** [comment]
Well I'll need to add a dummy then 
image

**Comment by louis-bompart** [comment]
> Well I'll need to add a dummy then
> image

Yep, do a dummy, remove it in the PR that include the first FT

**Comment by jeremierobert-coveo** [comment]
Done

**Comment by louis-bompart** [comment]
Why not using TypeScript tests?
https://docs.cypress.io/guides/tooling/typescript-support.html#Configure-tsconfig-json
https://github.com/bahmutov/add-typescript-to-cypress

It could allow us to refer to some code from our src easily (Like the classnames)

**Comment by jeremierobert-coveo** [comment]
Here some typescript sprinkle

File: README.md [hunk 3]
```diff
@@ -37,11 +37,18 @@ The code is written in [typescript](http://www.typescriptlang.org/) and compiled
...
 
 -   `npm run test` will execute the tests one time and give you the report
 -   `npm run watchTest` will watch changes and reexecute the tests and coverage when saving a file.
+
+## Running functional tests
+
+1. `npm run dev` will start the developer server.
+2. `npx cypress run` will run the cypress functional tests suite.
+
+To add a new test suite simply add a new spec file in `./cypress/integration/` folder.
```
Comment (nathanlb) [inline README.md]: "in **the** `./cypress/integration/` folder." would technically be better :sorry:

File: pages/attached_result.html [hunk 4]
```diff
@@ -38,16 +38,13 @@
                         getValues: function() {
                             return attachedIds;
                         }
-                    },
```
Comment (louis-bompart) [inline pages/attached_result.html]: Why is it modified?

Comment (jeremierobert-coveo) [reply pages/attached_result.html]: I removed the stuff that was there to test user actions

Comment (jeremierobert-coveo) [reply pages/attached_result.html]: I removed the stuff that was there to test user actions

**Pull request merged**